paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java

```diff
@@ -272,10 +272,14 @@ public void dropTable(Identifier identifier, boolean ignoreIfNotExists)
         try {
             getTable(identifier);
         } catch (TableNotExistException e) {
+            if (!tableExistsInFileSystem(getTableLocation(identifier), DEFAULT_MAIN_BRANCH)) {
```

Comment:
As far as I know, before Spark calls `catalog.dropTable`, it actually invokes `catalog.getTable`. It has already been skipped there.